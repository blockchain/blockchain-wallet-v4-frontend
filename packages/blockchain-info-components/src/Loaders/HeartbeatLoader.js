import React from 'react'
import PropTypes from 'prop-types'
import styled, { css, keyframes } from 'styled-components'
import { keysIn } from 'ramda'

import { Palette } from '../Colors'

const bounceFrames = keyframes`
  0%, 100% { transform: scale(0.0); }
  50% { transform: scale(1.0); }
`
const bounceAnimation = css`
  ${bounceFrames} 2s infinite ease-in-out;
`
const Container = styled.div`
  position: relative;
  width: ${props => props.width};
  height: ${props => props.height};
`
const Circle1 = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  border-radius: 50%;
  opacity: 0.7;
  background-color: ${props => props.theme[props.color]};
  animation: ${bounceAnimation};
`
const Circle2 = styled(Circle1)`
  animation-delay: -1s;
`

const HeartbeatLoader = props => {
  const { color, ...rest } = props

  return (
    <Container {...rest}>
      <Circle1 color={color} />
      <Circle2 color={color} />
    </Container>
  )
}

HeartbeatLoader.propTypes = {
  width: PropTypes.string.isRequired,
  height: PropTypes.string.isRequired,
  color: PropTypes.oneOf(keysIn(Palette()))
}

HeartbeatLoader.defaultProps = {
  width: '40px',
  height: '40px',
  color: 'brand-secondary'
}

export default HeartbeatLoader
