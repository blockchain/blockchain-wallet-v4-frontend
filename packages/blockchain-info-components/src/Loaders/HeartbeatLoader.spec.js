import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import HeartbeatLoader from './HeartbeatLoader'

describe('HeartbeatLoader component', () => {
  it('default renders correctly', () => {
    const component = shallow(
      <HeartbeatLoader height='10px' width='10px' color='brand-primary' />
    )
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
