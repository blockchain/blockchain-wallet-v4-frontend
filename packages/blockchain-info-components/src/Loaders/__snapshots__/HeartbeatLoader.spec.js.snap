// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeartbeatLoader component default renders correctly 1`] = `
<HeartbeatLoader__Container
  height="10px"
  width="10px"
>
  <HeartbeatLoader__Circle1
    color="brand-primary"
  />
  <HeartbeatLoader__Circle2
    color="brand-primary"
  />
</HeartbeatLoader__Container>
`;
